python/tests/expressions/test_evaluator.py

```diff
@@ -52,112 +55,126 @@
 )
 
 
+def _record_simple(id: int, data: Optional[str]) -> Record:
+    r = Record(SIMPLE_SCHEMA.as_struct())
```

Comment:
Overall, I'm okay with the changes and I approved this PR. But I think it's a little awkward to need to construct `Record` instances this way.

I think we may be able to do both if we make `Record.__init__` like this:

```python
def Record:
    def __init__(self, *data, struct=None):
        self.struct = struct
        size = len(struct) if struct else len(data)
        self.data = list([None] * size)
        for pos, value in enumerate(data):
            self.data[pos] = value
```

That allows passing `struct` as a keyword arg, or passing raw values so both of these work correctly:

```python
Record(1, "a")
Record(struct=StructType(...))
```

Reply:
I removed it because we only constructed it in that way in the tests. I've re-added it slightly differently:
```python
def test_record_positional_args() -> None:
    r = Record(1, "a", True)
    assert repr(r) == "Record[field1=1, field2='a', field3=True]"
```
This is what I ended up with:
```python
    def __init__(self, *data: Any, struct: Optional[StructType] = None) -> None:
        if struct is not None:
            self._position_to_field_name = {idx: field.name for idx, field in enumerate(struct.fields)}
        elif data is not None:
            self._position_to_field_name = {}
            for idx, d in enumerate(data):
                self._position_to_field_name[idx] = f"field{idx + 1}"
                self[idx] = d
```

This way we don't have the awkward if's where we have to check if it is positional or keyword based. WDYT?